src/RA_Core.cpp

```diff
@@ -1629,35 +1629,39 @@ BOOL _FileExists(const std::string& sFileName)
     }
 }
 
+
+
 std::string GetFolderFromDialog()
 {
     std::string sRetVal;
-    //HRESULT hr = CoInitializeEx( nullptr, COINIT_APARTMENTTHREADED|COINIT_DISABLE_OLE1DDE );
-    IFileOpenDialog* pDlg = nullptr;
-    HRESULT hr = CoCreateInstance(CLSID_FileOpenDialog, nullptr, CLSCTX_ALL, IID_IFileOpenDialog, reinterpret_cast<void**>(&pDlg));
-    if (hr == S_OK)
+
+
+	CComPtr<IFileOpenDialog> pDlg;
+	if (auto hr = CoCreateInstance(CLSID_FileOpenDialog, nullptr, CLSCTX_ALL, IID_IFileOpenDialog, 
+		reinterpret_cast<void**>(&pDlg)); SUCCEEDED(hr))
     {
         pDlg->SetOptions(FOS_PICKFOLDERS);
-        hr = pDlg->Show(nullptr);
-        if (hr == S_OK)
+
+		if (hr = pDlg->Show(nullptr); SUCCEEDED(hr))
```

Comment:
Please separate this into two lines as it was previously. It's unusual to put a semicolon in the middle of a conditional clause.

Last time I suggested this you mentioned minimizing the scope. Having two separate lines doesn't change the scope of anything. `hr` is defined outside the if statement.

If you're that concerned about putting everything on one line, consider this:
```
if (SUCCEEDED(hr = pDlg->Show(nullptr)))
```

Reply:
The selection statement keep scopes tight. It's not unusual you just don't know about it, everything you don't know is strange at first. People who don't know programming at all think code looks like gibberish.
http://en.cppreference.com/w/cpp/language/if

What you suggested seems like a good middle ground though.